FakeServer/GraphQL/GraphQLMiddleware.cs

```diff
@@ -75,5 +91,28 @@ public GraphQLMiddleware(RequestDelegate next, IDataStore datastore, IMessageBus
 
             await context.Response.WriteAsync(json);
         }
+
+        private static async Task<string> ParseQuery(HttpContext context)
```

Comment:
Using Exceptions as a control flow is sometimes a good solution, but here as we handle it already in the caller and there is only 1 exception case, I would recommend to use value tuples.

```cs
private static async Task<(bool success, string body)> ParseQuery(HttpContext context)

...

return (true, body);
...
return (false, null);
...
return (true, jsonBody.query);
```

In real life when these get more complicated it is recommended to use Class as a return value, so also error message etc. can be added as properties.

Reply:
Thanks for the instructive feedback. Agree with all you wrote, but here aren't there two exception cases: json deserialization failure and missing `query` property in json? Shouldn't it be better then to use either exceptions or custom class with error message here, as you said?